src/graph/query_graph.c

```diff
@@ -222,6 +228,55 @@ EntityType QueryGraph_GetEntityTypeByAlias(const QueryGraph *qg, const char *ali
 	return ENTITY_UNKNOWN;
 }
 
+// Returns if there was an update for an unkown label id.
+static bool _QGNode_Update(QGNode *node) {
+	bool updated = false;
+	if(node->labelID == GRAPH_UNKNOWN_LABEL) {
+		GraphContext *gc = QueryCtx_GetGraphCtx();
+		Schema *s = GraphContext_GetSchema(gc, node->label, SCHEMA_NODE);
+		if(s) {
+			node->labelID = s->id;
+			updated = true;
+		}
+	}
+	return updated;
+}
+
+// Returns the amount of updates to unkowns relationship types of the query graph edge.
+static uint _QGEdge_Update(QGEdge *edge) {
+	uint updates = 0;
+	GraphContext *gc = NULL;
+	uint rel_types_count = array_len(edge->reltypeIDs);
+	for(uint i = 0; i < rel_types_count; i++) {
+		if(edge->reltypeIDs[i] == GRAPH_UNKNOWN_RELATION) {
+			gc = gc ? gc : QueryCtx_GetGraphCtx();
+			Schema *s = GraphContext_GetSchema(gc, edge->reltypes[i], SCHEMA_EDGE);
+			if(s) {
+				edge->reltypeIDs[i] = s->id;
+				updates++;
+			}
+		}
+	}
+	return updates;
+}
+
+void QueryGraph_Update(QueryGraph *qg) {
+	// No unknown labels or relationships - no need to updated.
+	if(qg->unkown_label_ids == 0) return;
+	uint node_count = QueryGraph_NodeCount(qg);
+	uint edge_count = QueryGraph_EdgeCount(qg);
+
+	// Update nodes.
+	for(uint i = 0; i < node_count; i++) {
+		if(_QGNode_Update(qg->nodes[i])) qg->unkown_label_ids--;
```

Comment:
`_QGNode_Update` should acts in a similar way to `_QGEdge_Update`
Either:
```
if(_QGNode_Update(qg->nodes[i])) qg->unkown_label_ids--;
if(_QGEdge_Update(qg->edges[i])) qg->unkown_label_ids--;
```
Or 
```
qg->unkown_label_ids -= _QGNode_Update(qg-> nodes[i]);
qg->unkown_label_ids -= _QGEdge_Update(qg->edges[i]);
```